### PR TITLE
Types for culori

### PR DESCRIPTION
**Pull request by bijela-gora**
Link to the culori lib: https://github.com/Evercoder/culori

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
  - the type definitions was tested in [oklch-picker](https://github.com/evilmartians/oklch-picker) project with following steps:
    - manually remove `declare module 'culori/fn'` section in the `oklch-picker/types.d.ts`
    - run `pnpm add @types/culori@file:../DefinitelyTyped/types/culori/`
    - run `pnpm run test`
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
  - the package author decided to not include .d.ts files in the culori repo https://github.com/Evercoder/culori/pull/166#issuecomment-1435283290
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
  - can't be done because of https://github.com/microsoft/dts-gen/issues/156 
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
Please add a test file, with just proof of working usage, say module (should be there, when using generator script). The lib is too large to just skip this step IMO, thanks!

File: types/culori/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        "baseUrl": "../",
+        "typeRoots": ["../"],
+        "types": [],
+        "target": "es6",
+        "strict": true,
+        "forceConsistentCasingInFileNames": true,
+        "exactOptionalPropertyTypes": true,
+        "noUncheckedIndexedAccess": true,
+        "noUnusedLocals": false,
+        "noUnusedParameters": true
```
Comment (peterblazejewicz) [inline types/culori/tsconfig.json:15]:
```suggestion
        "module": "commonjs",
        "lib": [
            "es6"
        ],
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strictFunctionTypes": true,
        "strictNullChecks": true,
        "baseUrl": "../",
        "typeRoots": [
            "../"
        ],
        "types": [],
        "noEmit": true,
        "forceConsistentCasingInFileNames": true
```
those are non-standard settings, this is why warning label was added. Why do you want those different settings? thx!

**Comment by bijela-gora** [comment]
@peterblazejewicz 

Hello!

Thanks for the quick reply. As I said in the description, the main tests are in [a separate repository](https://github.com/bijela-gora/Tests-for-culori-types). I copied some of them to [types/culori in this repo](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64686/commits/cc7fa625d0e77f7c997cde8c3d8c0743a5dfe700)

If you know of a better way to do this, I'd be happy to hear. The only thing is that in the unhappy path tests I would like to be able to set expectations for specific error messages, like here https://github.com/bijela-gora/Tests-for-culori-types/blob/trunk/tests/average.errors.ts#L3.

File: types/culori/tsconfig.json (hunk 1 above)
Comment (bijela-gora) [reply types/culori/tsconfig.json:15]: I want to make checks of TS compiler as strict as possible. All this compilerOptions I copied from https://github.com/tsconfig/bases/blob/main/bases/node18-strictest.combined.json

Comment (bijela-gora) [reply types/culori/tsconfig.json:15]: @peterblazejewicz 

I checked that most of the options I added don't really add any value. But I can't decide whether `exactOptionalPropertyTypes` is useful or not. What do you think about it?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@bijela-gora thanks!

**Comment by bijela-gora** [comment]
@peterblazejewicz I added more tests.

File: types/culori/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,282 @@
+// Type definitions for culori 2.0
+// Project: https://github.com/Evercoder/culori
+// Definitions by: Bijela Gora <https://github.com/bijela-gora>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="./fn/index.d.ts" />
```
Comment (peterblazejewicz) [inline types/culori/index.d.ts:6]: so this line is here just to make sure `culori/fn` symbols are exported?
If so, you can consider removing this one, adding 'OTHER_FILES.TXT' file and referencing `fn/index.d.ts` in it:
https://github.com/DefinitelyTyped/DefinitelyTyped#other_filestxt

Comment (bijela-gora) [reply types/culori/index.d.ts]: I just removed `/// <reference path="./fn/index.d.ts" />` line.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@bijela-gora thanks!

**Comment by bijela-gora** [comment]
Ready to merge

**Comment by Jeremy-Knudsen** [comment]
Great to see this update, but one thing I see missing is support for the added color space XYB which is a perceptual color space used in JPEG XL. I was hoping this update would have included it. Any chance it could be added?

https://culorijs.org/api/#color-spaces

<html>
<body>


xyb | XYB color space | modeXyb
-- | -- | --



</body>
</html>

**Comment by Jeremy-Knudsen** [comment]
> Great to see this update, but one thing I see missing is support for the added color space XYB which is a perceptual color space used in JPEG XL. I was hoping this update would have included it. Any chance it could be added?
> 
> https://culorijs.org/api/#color-spaces
> 
> xyb	XYB color space	modeXyb

@bijela-gora Is there any chance you can add the XYB color space that is new to culori to the type definitions? :) @mentioning you in case you didn't see my message.

**Comment by bijela-gora** [comment]
@Jeremy-Knudsen hi

I have no plans to support this types definitions. Adding types for the culori lib was résumé-driven project. I am open to transferring ownership.

**Pull request merged**